Drop pandas dependencies + remove by test created files

gnzng opened the pull request:
As mentioned in #545, this PR drops pandas.

1dd740f175a25f1628b451e2f90bb9f06e6da73c substitutes the `remove_spikes_pandas()`  with a `numpy` version: `remove_spikes_numpy()` in `math/deglitch.py`. Added a test for this as well. The result works quite well:

<img width="871" alt="numpy" src="https://github.com/user-attachments/assets/c560d522-9351-47b3-adda-487e047a68a1" />

b2ddb0d24867699c381631dbca4a3a7c362cdf4e remove the pandas dependencyin `xrd/instruct2xas.py`. 

a6094a796de7f5b064617636358708661ace9565 and 0c449c737fc2ee5c7b56039beccb1b5b408ba68c remove the by `pytest` created test files. 

0c449c737fc2ee5c7b56039beccb1b5b408ba68c is just a stylistic cleanup. 

All tests are passing on `platform darwin -- Python 3.12.0, pytest-8.3.4`

maurov commented:
@gnzng thanks! removing Pandas from `struct2xas` and using numpy for spikes removal is fine for me. I just have few recommendations for the other commits:

- keep `import numpy.testing` -> this is required for running automatic tests
- prefer using `from pathlib import Path` instead of `os`

Please, could you change your code following these recommendations? Otherwise I can do it by myself before merging.

gnzng commented:
@maurov thanks a lot! 

here is the updated branch with the updates

I kept `Path` and `numpy.testing`, [df943a1](https://github.com/xraypy/xraylarch/pull/550/commits/df943a1f56c1976114f442b3e4b7ed623e7a2755) and [8397e7f](https://github.com/xraypy/xraylarch/pull/550/commits/8397e7f732a48ad4c23d3e42f03a81fb4bb4e494)

In [8f9aa19](https://github.com/xraypy/xraylarch/pull/550/commits/8f9aa1949ee768ea0766c58f8a6e26fd2755484a), I also added a `numpy`-only function, since it was actually using `scipy.ndimage.median_filter` under the hood, but kept the `scipy` version as `remove_spikes_scipy()`. Added tests in [34ff624](https://github.com/xraypy/xraylarch/pull/550/commits/34ff62471d3582737339f45a28819d4b4f3b279c). 

<img width="870" alt="np_scp" src="https://github.com/user-attachments/assets/c9f6a5f6-cfdb-4dae-bc3d-b5988e8deace" />

from benchmarking the `scipy' version beats the pure `numpy` version:
```shell
numpy took 0.579102 seconds
scipy took 0.001220 seconds
```

newville commented:
@gnzng Thanks - for de-glitching and struct2xas this looks really nice.  And thanks for the numpy vs scipy test -- a clear winner!

For testing: yes, there are many legacy things here, and the testing is generally "poor".   
"unittest" is built-in but very old, but not going away anytime soon.  We could migrate away from it.

We are testing with `pytest` - and it is better than unittest. It is OK to import some of its modules and features in the tests. This does not always seem necessary, but it's fine to use.    

`numpy.testing` adds a lot of value for working with ndarrays, and is also OK to use.

I'll also say here, that I have grown to like `pathlib` quite a bit, and am +1 on `pathlib.Path.unlink` vs `os` or `shutil` for removing files.

I'm OK with merging this if you agree.

maurov commented:
@gnzng thanks!

@newville looks fine to me now, feel free to merge. And yes, let's keep testing with `pytest` and `numpy.testing`. We may had to define a deployment workflow for automatically run the test at each pull request.

newville commented:
@gnzng Thanks again!   
@maurov Ah, good point -- CI tests with GitHub Actions are not being run on pull requests.  Will fix.
